examples/AffinePlaneCurve.jl

```diff
@@ -384,14 +396,229 @@ function curve_singular_locus(C::AffinePlaneCurve)
+  if P.ambient_dim != 2
+     error("The point needs to be in a two dimensional space")
+  end
+  D = curve_map_point_origin(C, P)
+  G = D.eq
+  R = parent(G)
+  V = gens(R)
+  A = grade(R)
+  HC = homogenous_components(A(G))
+  L = collect(keys(HC))
+  M = sort(L, by=_sort_helper_multiplicity)
+  Gm = HC[M[1]]
+  Z = factor(Gm)
+  D = Dict{AffinePlaneCurve{S}, Int64}()
+  X = V[1] - P.coord[1]
+  Y = V[2] - P.coord[2]
+  for p in keys(Z.fac)
+     if total_degree(p.f) == 1
+        push!(D, AffinePlaneCurve(evaluate(p.f, [X, Y])) => Z.fac[p])
+     end
+  end
+  return(D)
+end
+
+################################################################################
+# check for equality of curves (the equations are equal up to multiplication by
+# a non zero constant).
+
+function ==(C::AffinePlaneCurve, D::AffinePlaneCurve)
+  F = C.eq
+  G = D.eq
+  d = div(F, G)
+  if isconstant(d) == false
+     return false
+  else
+     if F - d*G == 0
+        return true
+     else
+        return false
+     end
+  end
+end
+
+################################################################################
+# Singular locus in the sense of varieties.
+# Might change depending on the futur changes on VarietyModule.
+
+@doc Markdown.doc"""
+singular_locus(C::AffinePlaneCurve)
+> Returns the singular locus of C as a variety.
+"""
+function singular_locus(C::AffinePlaneCurve)
+  J = jacobi_ideal(C)
+  R = parent(C.eq)
+  S = ideal(R, [gens(J); C.eq])
+  return Variety(S)
+end
+
+################################################################################
+# Compute the intersection multiplicity of the two affine plane curves at the
+# given point.
+
+@doc Markdown.doc"""
+intersection_multiplicity(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+> Returns the intersection multiplicity of C and D at P.
+"""
+function intersection_multiplicity(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+  if P.ambient_dim != 2
+     error("The point needs to be in a two dimensional space")
+  end
+  R = parent(C.eq)
+  m = ideal_point(R, P)
+  r = Localization(R, m)
+  I = ideal(r, [C.eq, D.eq])
+  groebner_basis(I)
+  return(Singular.vdim(I.gb.S))
+end
+
+################################################################################
+# Check if the two curves intersect transversally at the given point.
+
+@doc Markdown.doc"""
+aretransverse(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}, P::Point{S}) where S<:FieldElem
+> Returns true if C and D intersect transversally at P and false otherwise.
+"""
+function aretransverse(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}, P::Point{S}) where S<:FieldElem
+  if P.ambient_dim != 2
+     error("The point needs to be in a two dimensional space")
+  end
+  if issmooth(C, P) == false || issmooth(D, P) == false
+     return false
+  else
+     if intersection_multiplicity(C, D, P) == 1
+        return true
+     else
+        return false
+     end
+  end
```

Comment:
This is probably somewhat subjective, but I personally find the following alternative easier to understand, and of course it is shorter as well
```suggestion
  return issmooth(C, P) && issmooth(D, P) && intersection_multiplicity(C, D, P) == 1
```